src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -104,8 +108,15 @@ public class AddressIndexer : IAddressIndexer
 
         private Task<ChainedHeaderBlock> prefetchingTask;
 
-        /// <summary>Maximum supported reorganization length.</summary>
-        private readonly int maxReorgLength;
+        /// <summary>Distance in blocks from consensus tip at which compaction should start.</summary>
+        /// <remarks>It can't be lower than maxReorg since compacted data can't be converted back to uncompacted state for partial reversion.</remarks>
+        private readonly int compactionTriggerDistance;
+
+        /// <summary>
+        /// This is a window of some blocks that is needed to reduce the consequences of nodes having different view of consensus chain.
+        /// We assume that nodes usually don't have view that is different from other nodes by that constant of blocks.
+        /// </summary>
+        public const int SyncBuffer = 50;
```

Comment:
Hmm, trying to understand what this is for? I see that it goes into compactionTriggerDistance. Why would `maxReorg * 2` not be enough on its own?

Reply:
Because we need a floating window of 50 blocks and if your node is behind or ahead by less than half of that window it will still be ok for you to operate and reach consensus on sidechain.

Reply:
So one guy creates a block with commitment of `consensusTip.Height - MaxReorg - syncDiff`
Another guy has to have data at `Mainchain consensus tip height - MaxReorg` and blow. So effectively guy that checks the block can be behind by `SyncDiff` blocks.

Also if that guy that is behind produces a block and he is behind it's ok, others can all validate it, but not if he is behind more than `MaxReorg*2 - syncDiff` since after that we will have compaction.